storage/backend.go

```diff
@@ -323,6 +324,35 @@ func (b *Backend) RemoveVolume(vol *Volume) error {
 	return nil
 }
 
+func (b *Backend) CreateVolumeSnapshot(snapshotName string, volConfig *VolumeConfig) (*Snapshot, error) {
+
+	log.WithFields(log.Fields{
+		"backend":       b.Name,
+		"storage_class": volConfig.StorageClass,
+		"source_volume": volConfig.Name,
+		"snapshot_name": snapshotName,
+	}).Debug("Attempting snapshot create.")
+
+	// Prepare volume
+	if volConfig.InternalName == "" {
+		volConfig.InternalName = b.Driver.GetInternalVolumeName(volConfig.Name)
+	}
+
+	// Create snapshot
+	snapshot, err := b.Driver.CreateSnapshot(snapshotName, volConfig)
+	if err != nil {
+		return nil, err
+	}
+	// Update snapshot with relevant backend information
+	vol, ok := b.Volumes[volConfig.Name]
```

Comment:
You don't need this block.  Pool isn't needed at all, and you already have the backend name (b.Name).